### ISSUE TITLE
LB4 CouchDB - Document update conflict. (duplicate?)

### ISSUE THREAD

**Issue by mapmalith** [strongloop/loopback#4212]
# Description/Steps to reproduce
Create New Model as below,
`@property({
    type: 'string', 
    id: true, }) 
id?: string;

@property({
    type: 'string',
    required: true,
    jsonSchema: {
      "format": "email"
    }
  })
  email: string;

  @property({
    type: 'string',
    required: true,
  })
  password: string;

  @hasOne(() => Profile)
  profile?: Profile;

And then created a CouchDB Data-source, Repository & a CRUD Controller for the model by using the scaffolding way.

When I try to PUT method as below,
`localhost:3000/users/2151624a667217dfb9361631c0000879`

with passing the below raw body,
`{
  "id": "2151624a667217dfb9361631c0000879",
  "email": "userrrr@example.com",
  "password": "string"
}`

it returns the following `409` error
`{
    "error": {
        "statusCode": 409,
        "name": "Error",
        "message": "Document update conflict. (duplicate?)"
    }
}`

By searching I came to know that this because of  CouchDB "_rev" conflict and would like to know how this is handled within the framework?

TIA.

# Link to reproduction sandbox

# Expected result

# Additional information

**Comment by dhmlau** [comment]
@jannyHou, could you PTAL? Thanks.

**Comment by jannyHou** [comment]
looking

**Comment by jannyHou** [comment]
@mapmalith Can you try remove `"id": "2151624a667217dfb9361631c0000879"` in the payload?

Since you are calling the endpoint that maps to "update by id", the payload should exclude the id fields.

**Comment by mapmalith** [comment]
@jannyHou thanks for your response :)

By default when we create a controller including CRUD operations,
`@put('/users/{id}'` uses the `replaceById` function and 
`@patch('/users/{id}'` uses the `updateById` function

Further I have tried the payload by removing `"id": "2151624a667217dfb9361631c0000879"` from the payload and still getting the same error - `{ "error": { "statusCode": 409, "name": "Error", "message": "Document update conflict. (duplicate?)" } }`

To examine I changed the `replaceById` to `updateById` in the `@put('/users/{id}'` and it works with & without `"id": "2151624a667217dfb9361631c0000879"` in the payload.

Thanks.
